indexer/indexer.go

```diff
@@ -1082,28 +1082,76 @@ func (collection *Collection) BestIndex(ctx context.Context, query Query) (*Inde
 	var best *Index
 	bestMatchingFields := 0
 
-	for _, index := range indexes {
-		// Iterate over the fields
+	for idx, index := range indexes {
 		matchingFields := 0
-		for _, field := range index.fields {
+		if query.Sort != "" {
+			contain := false
```

Comment:
Could you move this into the loop on line 1101 to reduce the number of times we iterate?

Maybe have a `sawSort := false` and set it to true on line 1103, then on 1118 continue if we needed sort but didn't see it.

Reply:
Ok, I have fixed it